homeassistant/components/sensor/batinfo.py

```diff
@@ -0,0 +1,115 @@
+"""
+Details about the built-in battery.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.batinfo/
+"""
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_NAME
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['batinfo==0.3']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_NAME = 'name'
+ATTR_PATH = 'path'
+ATTR_ALARM = 'alarm'
+ATTR_CAPACITY = 'capacity'
+ATTR_CAPACITY_LEVEL = 'capacity_level'
+ATTR_CYCLE_COUNT = 'cycle_count'
+ATTR_ENERGY_FULL = 'energy_full'
+ATTR_ENERGY_FULL_DESIGN = 'energy_full_design'
+ATTR_ENERGY_NOW = 'energy_now'
+ATTR_MANUFACTURER = 'manufacturer'
+ATTR_MODEL_NAME = 'model_name'
+ATTR_POWER_NOW = 'power_now'
+ATTR_SERIAL_NUMBER = 'serial_number'
+ATTR_STATUS = 'status'
+ATTR_VOLTAGE_MIN_DESIGN = 'voltage_min_design'
+ATTR_VOLTAGE_NOW = 'voltage_now'
+
+DEFAULT_NAME = 'Battery'
+DEFAULT_PATH = '/sys/class/power_supply'
+
+ICON = 'mdi:battery'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the BatInfo sensor."""
+    name = config.get(CONF_NAME)
+
+    if not os.listdir(DEFAULT_PATH):
+        _LOGGER.error("No battery found")
+        return False
+
+    add_devices([BatInfoSensor(name)])
```

Comment:
I doubt that people will use HA on their old Lenovo T-series machines but you never know, right? Instead of adding all battery I made it configurable.